### PR TITLE
Initialize back end directory structure and create API endpoints that return dummy data.

### LINKED ISSUE #5: Firebase SDK Dashboard: Directory Structure
Initialize the directory structure in the github repo

### LINKED ISSUE #12: Create Dummy API endpoints
Create the api endpoints that the front end will interact with and make them return dummy data.

### PR DESCRIPTION
**Pull request by eriyezab**
Found the proper dependencies for jersey API.
Initialized the back end directory structure.
Created all p1 endpoints and configured them to return dummy data.
Created a .gitignore file and removed all *.swp files as well as the entire target/ directory from the repository.

### CODE REVIEW

File: src/main/java/com/google/capstone/api/MyResource.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Root resource (exposed at "myresource" path)
+ */
+@Path("myresource")
+public class MyResource {
```
Comment (shrutip90) [inline src/main/java/com/google/capstone/api/MyResource.java]: Looks like this is an example endpoint. Do we need to check this in or can we skip it?

**Review by shrutip90 — CHANGES_REQUESTED** [review]
nit: Change the PR description to say 'enable Jersey API'

File: src/main/java/firebase/sdk/dashboard/data/User.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package firebase.sdk.dashboard.data;
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/data/User.java]: I hope this User class is outdated & #18  has the latest version. Hence I am skipping the review of User class in this PR.

File: src/main/java/firebase/sdk/dashboard/data/SDKRelease.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package firebase.sdk.dashboard.data;
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/data/SDKRelease.java]: Skipping the review here and will do in #18.

File: src/main/java/firebase/sdk/dashboard/data/SDK.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package firebase.sdk.dashboard.data;
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/data/SDK.java]: Skipping the review here and will do in #18.

File: src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDao.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
+package firebase.sdk.dashboard.dao;
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDao.java]: Skipping the review of Model objects & Dao classes here and will do in #18.

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class UserResource {
+
+  private ArrayList<String> favourites;
+  public UserResource() {
+    favourites = new ArrayList<String>(Arrays.asList("firebase-common", "firebase-common-ktx", "firebase-installations", "firebase-installations-interop"));
+  }
+
+  //private static final PlatformReleaseDao PRDAO = new PlatformReleaseDaoDatastore();
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java]: Delete this comments if not required else add a TODO on when you plan tp use this

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  @POST
+  @Path("{uid}/sdks")
+  public Response addSdkToUserFavorites(@PathParam("uid") String uid) {
+    // TODO: implement this method
+    return ResponseHandler.createJsonResponse(Status.OK, favourites);
+  }
+
+  @DELETE
+  @Path("{uid}/sdks/{sdkName}")
+  public Response addSdkToUserFavorites(@PathParam("uid") String uid, @PathParam("sdkName") String sdkName) {
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java]: method names?

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+   *
+   * @return String that will be returned as a text/plain response.
+   */
+  @POST
+  @Path("{uid}/sdks")
+  public Response addSdkToUserFavorites(@PathParam("uid") String uid) {
+    // TODO: implement this method
+    return ResponseHandler.createJsonResponse(Status.OK, favourites);
+  }
+
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java:60]: Javadocs

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+   */
+  @GET
+  @Path("{uid}/sdks")
+  public Response getUserFavorites(@PathParam("uid") String uid) {
+    // TODO: implement this method
+    return ResponseHandler.createJsonResponse(Status.OK, favourites);
+  }
+
+  /**
+   * Method handling HTTP GET requests. The returned object will be sent
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java]: Update the javadocs

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+   * to the client as "text/plain" media type.
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java]: Response is sent as json vs plain text. Please take a look.

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+import firebase.sdk.dashboard.dao.SDKDao;
+
+/**
+ * Root resource (exposed at "myresource" path)
+ */
+@Path("platforms")
+@Produces(MediaType.APPLICATION_JSON)
+public class PlatformResource {
+
+  //private static final PlatformReleaseDao PRDAO = new PlatformReleaseDaoDatastore();
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: If comments required add a TODO

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+//  private static final SDKDao SDAO = new SDKDao();
+
+  /**
+   * Method handling HTTP GET requests. The returned object will be sent
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: I think the Javadoc will change when you DAO & Data objects. Add a TODO so that you dont forget

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+      Date date = new Date();
+      HashMap<String, Long> platforms = new HashMap<String, Long>();
+      platforms.put("Android", date.getTime());
+      platforms.put("iOS", date.getTime());
+      platforms.put("Web", date.getTime());
+      platforms.put("Games", date.getTime());
+      return ResponseHandler.createJsonResponse(Status.OK, platforms);
+  }
+
+  @Path("/{platform}/releases")
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java:49]: Annotation to say what type of call is this (GET/POST etc.)?

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+  public ReleaseResource getReleaseResource(@PathParam("platform") String platform) {
+    return new ReleaseResource(platform);
+    }
+
+  @Path("/{platform}/sdks")
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java:63]: When will this resource be used?

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+    sdks.add("firebase-components");
+  }
+
+  /**
+   * Method handling HTTP GET requests. The returned object will be sent
+   * to the client as "text/plain" media type.
+   *
+   * @return String that will be returned as a text/plain response.
+   */
+  @GET
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java:65]: Path?

File: src/main/java/firebase/sdk/dashboard/api/SDKResource.java [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * Root resource (exposed at "myresource" path)
+ */
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class SDKResource {
+
+  private String platform;
+  private static final PlatformReleaseDao PRDAO = new PlatformReleaseDaoDatastore();
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/SDKResource.java]: Descriptive constant names.

File: src/main/java/firebase/sdk/dashboard/api/SDKResource.java [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+    this.platform = platform;
+  }
+
+  /**
+   * Method handling HTTP GET requests. The returned object will be sent
+   * to the client as "text/plain" media type.
+   *
+   * @return String that will be returned as a text/plain response.
+   */
+  @GET
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/SDKResource.java:41]: path?

File: src/main/java/firebase/sdk/dashboard/api/SDKResource.java [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+    }*/
+  }
+
+  @GET
+  @Path("/{sdkName}")
+  public Response getSDK(@PathParam("sdkName") String sdkName) {
+    SDK sdk = new SDK(sdkName, "firebase-common", "firebase-common");
+    sdk.setOwner("ashwin@");
+    int release = 78;
+    for (int i = 0; i < 5; i++) {
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/SDKResource.java]: Why are we doing this?

**Review by ankitaj224 — CHANGES_REQUESTED** [review]

File: src/main/java/firebase/sdk/dashboard/api/SDKResource.java (hunk 18 above)
Comment (eriyezab) [reply src/main/java/firebase/sdk/dashboard/api/SDKResource.java]: This is generating dummy data for the version history page.

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java (hunk 15 above)
Comment (eriyezab) [reply src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java:65]: This endpoint doesn't require an explicit path annotation because that is handled in the platform resource file. However I am adding the paths to the java docs of each endpoint method.

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java (hunk 14 above)
Comment (eriyezab) [reply src/main/java/firebase/sdk/dashboard/api/PlatformResource.java:63]: when searching for sdks to add to favorites

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 19]
```diff
@@ -0,0 +1,68 @@
...
+   */
+  @GET
+  public Response getPlatforms() {
+    // TODO: implement this method
+      HashMap<String, Instant> platforms = new HashMap<String, Instant>();
+      platforms.put("Android", Instant.now());
+      platforms.put("iOS", Instant.now());
+      platforms.put("Web", Instant.now());
+      platforms.put("Games", Instant.now());
+      return ResponseHandler.createJsonResponse(Status.OK, platforms);
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: nit: Check indentation

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+@Produces(MediaType.APPLICATION_JSON)
+public class PlatformResource {
+
+  /**
+   * Method handling HTTP GET requests.
+   * Exposed at "v1/platforms", this endpoint returns a list
+   * of the names of all platforms along with the timestamp representing
+   * the closest deadline.
+   *
+   * @return Response object containing astatus code and a HashMap contaning
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: nit: astatus -> HTTP status

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 21]
```diff
@@ -0,0 +1,68 @@
...
+ */
+@Path("platforms")
+@Produces(MediaType.APPLICATION_JSON)
+public class PlatformResource {
+
+  /**
+   * Method handling HTTP GET requests.
+   * Exposed at "v1/platforms", this endpoint returns a list
+   * of the names of all platforms along with the timestamp representing
+   * the closest deadline.
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: nit: -> release deadline

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 22]
```diff
@@ -0,0 +1,68 @@
...
+      platforms.put("Android", Instant.now());
+      platforms.put("iOS", Instant.now());
+      platforms.put("Web", Instant.now());
+      platforms.put("Games", Instant.now());
+      return ResponseHandler.createJsonResponse(Status.OK, platforms);
+  }
+
+  /**
+   * Method handling HTTP GET requests.
+   * Exposed at "v1/platforms/{plstform}/releases", this endpoint returns
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: nit: spelling of platform

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 23]
```diff
@@ -0,0 +1,68 @@
...
+   * a ReleaseResource object which handles the functionality of all endpoints
+   * starting with the endpoint path.
+   *
+   * @return A ReleaseResource that will handle all requests from this enpoint and
+   * onwards.
+   */
+  @Path("/{platform}/releases")
+  public ReleaseResource getReleaseResource(@PathParam("platform") String platform) {
+    return new ReleaseResource(platform);
+    }
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: nit: indentation

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 24]
```diff
@@ -0,0 +1,175 @@
...
+      .oldVersion(oldVersion)
+      .additionalInfo(new HashMap<String, String>())
+      .build();
+    return ResponseHandler.createJsonResponse(Status.OK, version);
+  }
+  
+  /**
+   * Method handling HTTP DELETE requests.
+   * Exposed at "v1/platforms/{platform}/releases/{release}/sdks/{sdkName}", this endpoint
+   * disenrolls the given sdk from the release it is in and deletes it from the database.
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: nit: from the given release, if it is a part of it and deletes the entries from the database.

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 25]
```diff
@@ -0,0 +1,175 @@
...
+
+    return ResponseHandler.createJsonResponse(Status.OK, releases);
+  }
+
+  /**
+   * Method handling HTTP POST requests.
+   * Exposed at "v1/platforms/{platform}/releases", this endpoint consumes a  
+   * Release object and adds it to the database. Only admins can do this.
+   *
+   * @return Response object containing a status code and a message to be
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: Do we need to send this message from server side? Looks like it can be derived from the status code.

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 26]
```diff
@@ -0,0 +1,175 @@
...
+    return ResponseHandler.createJsonResponse(Status.OK, message);
+  }
+
+  /**
+   * Method handling HTTP DELETE requests.
+   * Exposed at "v1/platforms/{platform}/releases/{release}", this endpoint
+   * deletes the given Release object from the database. Only admins can do this.
+   *
+   * @return Response object containing a status code and a message to be 
+   * displayed to the client.
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: ditto

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 27]
```diff
@@ -0,0 +1,175 @@
...
+    String message = String.format("Deleting %s release from %s platform", release, platform);
+    return ResponseHandler.createJsonResponse(Status.OK, message);
+  }
+
+  /**
+   * Method handling HTTP GET requests.
+   * Exposed at "v1/platforms/{platform}/releases/{release}/sdks", this endpoint
+   * returns a list of the names of all skds enrolled in the given release.
+   *
+   * @return Response object containing a status code and a list of Strings representing
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: You seem to be sending a Hashmap in the implementation. Please correct the javadoc.

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 28]
```diff
@@ -0,0 +1,175 @@
...
+    // TODO: implement this method
+    HashMap <String, ArrayList<String>> releaseSDKs = new HashMap<>();
+    releaseSDKs.put(release, sdks);
+    return ResponseHandler.createJsonResponse(Status.OK, releaseSDKs);
+  }
+
+  /**
+   * Method handling HTTP POST requests.
+   * Exposed at "v1/platforms/{platform}/releases/{release}/sdks", this endpoint 
+   * consumes an SDKRelease object and enrolls it in the given release in the database.
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: nit: remove 'in the database'. This applies to all APIs, so its not really required in the comments.

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java (hunk 6 above)
Comment (shrutip90) [reply src/main/java/firebase/sdk/dashboard/api/UserResource.java]: I still see the commented code here. Maybe an issue with viewing the latest changes. Please double check.

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 29]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  //private static final PlatformReleaseDao PRDAO = new PlatformReleaseDaoDatastore();
+//  private static final SDKDao SDAO = new SDKDao();
+
+  /**
+   * Method handling HTTP GET requests. The returned object will be sent
+   * to the client as "text/plain" media type.
+   *
+   * @return String that will be returned as a text/plain response.
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java]: Please update the return value.

**Review by shrutip90 — CHANGES_REQUESTED** [review]

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 30]
```diff
@@ -0,0 +1,69 @@
...
+   * onwards.
+   */
+  @Path("/{platform}/releases")
+  public ReleaseResource getReleaseResource(@PathParam("platform") String platform) {
+    return new ReleaseResource(platform);
+  }
+
+  /**
+   * Method handling HTTP GET requests.
+   * Exposed at "v1/platforms/{plstform}/sdks", this endpoint returns
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: platform spelling.

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 31]
```diff
@@ -0,0 +1,175 @@
...
+   * Exposed at "v1/platforms/{platform}/releases", this endpoint consumes a  
+   * Release object and adds it to the database. Only admins can do this.
+   *
+   * @return Response object containing a status code.
+   */
+  @POST
+  public Response addRelease(Release release) {
+    // TODO: Implement this method.
+    String message = String.format("Added %s release to the dashboard.", "M79"/*release.releaseName*/);
+    return ResponseHandler.createJsonResponse(Status.OK, message);
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: Not sure if I asked this before, but why do we need to send this message in the Response. Can't we generate it based on the Status code on the frontend?

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 32]
```diff
@@ -0,0 +1,175 @@
...
+    return ResponseHandler.createJsonResponse(Status.OK, releases);
+  }
+
+  /**
+   * Method handling HTTP POST requests.
+   * Exposed at "v1/platforms/{platform}/releases", this endpoint consumes a  
+   * Release object and adds it to the database. Only admins can do this.
+   *
+   * @return Response object containing a status code.
+   */
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java:93]: Add TODO: Check membership in Firebase core team for allowing only admin access to this method.

File: src/main/java/firebase/sdk/dashboard/api/UserResource.java [hunk 33]
```diff
@@ -0,0 +1,74 @@
...
+          "firebase-common-ktx", 
+          "firebase-installations", 
+          "firebase-installations-interop"));
+  }
+
+  /**
+   * Method handling HTTP GET requests. 
+   * Returns a list of the names of the given users favorite sdks. 
+   *
+   * @return A Response object containing an HTTP status code and a list of string
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/UserResource.java]: nit: strings

**Review by shrutip90 — CHANGES_REQUESTED** [review]

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java (hunk 14 above)
Comment (ankitaj224) [reply src/main/java/firebase/sdk/dashboard/api/PlatformResource.java:63]: sg, thanks.

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java (hunk 31 above)
Comment (eriyezab) [reply src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: You did ask this before sorry about that

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java (hunk 13 above)
Comment (ankitaj224) [reply src/main/java/firebase/sdk/dashboard/api/PlatformResource.java:49]: ?

File: src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java [hunk 34]
```diff
@@ -0,0 +1,172 @@
...
+   *
+   * @return Response object containing a status code and a HashMap with the key being the
+   * release and the values being a list of strings representing the names of all the 
+   * sdks enrolled in the given release.
+   */
+  @GET
+  @Path("{release}/sdks")
+  public Response getReleaseSDKs(@PathParam("release") String release) {
+    // TODO: implement this method
+    HashMap <String, ArrayList<String>> releaseSDKs = new HashMap<>();
```
Comment (ankitaj224) [inline src/main/java/firebase/sdk/dashboard/api/ReleaseResource.java]: Why have a hashmap if we only send a response for one key. Use List<String> instead?

**Review by ankitaj224 — APPROVED** [review]

File: src/main/java/firebase/sdk/dashboard/api/SDKResource.java [hunk 35]
```diff
@@ -0,0 +1,93 @@
...
+      sdks.add("firebase-common-ktx");
+      sdks.add("firebase-ml");
+      sdks.add("firebase-database");
+      sdks.add("firebase-auth");
+      sdks.add("firebase-messaging");
+      
+      return ResponseHandler.createJsonResponse(Status.OK, sdks);
+    /*} catch (Exception e) {
+      return e.toString();
+    }*/
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/SDKResource.java]: Remove commented code.

File: src/main/java/firebase/sdk/dashboard/api/SDKResource.java [hunk 36]
```diff
@@ -0,0 +1,93 @@
...
+   * Method handling HTTP GET requests.
+   * Exposed at "v1/platforms/{platform}/sdks", this endpoint returns a list
+   * of the names of all the sdks enrolled in the given platform.
+   *
+   * @return Response object containing a list of sdk names in JSON.
+   */
+  @GET
+  public Response getSDKs() {
+  
+    //try {
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/SDKResource.java]: If you don't need it, you can just remove it for now.

File: src/main/webapp/js/script.js [hunk 37]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
...
+       
+        divElement.appendChild(document.createElement("HR"));
+
+        const deadline = release["launchDeadline"];
+        const textNode1 = document.createTextNode("Launch Deadline: " + getDate(1595424871)); 
+        divElement.appendChild(textNode1);
+
+        divElement.appendChild(document.createElement("HR"));
+
+        const coldfreeze = release["codeFreezeDate"];
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: cold -> code. This is probably from the other PRs that were merged into this branch, but could you please change this.

File: src/main/webapp/js/script.js [hunk 38]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
...
+        const textNode3 = document.createTextNode("Cold Freeze Date: " + getDate(1595424871));
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: ditto

File: src/main/webapp/js/script.js [hunk 39]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
     $('div.setup-panel div a.btn-success').trigger('click');
 });
 
+
+
+
+//Fetch data from API endpoint
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: Space after //

File: src/main/webapp/js/script.js [hunk 40]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
...
+       if (plat == doms[count]){
+        const divElement = document.getElementById(plat); 
+        const textNode = document.createTextNode(platforms[plat]); 
+        divElement.appendChild(textNode);
+       }
+       count+=1;
+    }
+}
+
+//Fetch release pages
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: ditto

File: src/main/webapp/js/script.js [hunk 41]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
...
+async function getReleases(platform) {
+   const response = await fetch(`/v1/platforms/${platform}/releases`);
+   const releases = await response.json();
+   
+    //Create text elements for each release detail
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: ditto

File: src/main/webapp/js/script.js [hunk 42]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
...
+
+        divElement.appendChild(document.createElement("HR"));
+
+        const launch = release["launchDate"];
+        const textNode4 = document.createTextNode("Launch Date: " + getDate(Date.now())); 
+        divElement.appendChild(textNode4); 
+    }
+}
+
+// Fetch release pages
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: Wrong comment

File: src/main/webapp/js/script.js [hunk 43]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
...
+    createNode("Library Group: " + libraryGroup, "#library-group");
+    createNode(owner, "#owner");
+}
+
+function getDate(time) {
+    var date = new Date(time);
+    return date;
+}
+
+
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: Extra blank line.

File: src/main/webapp/js/script.js [hunk 44]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
     $('div.setup-panel div a.btn-success').trigger('click');
 });
 
+
+
+
+//Fetch data from API endpoint
+ 
+// Fetch for landing page
+
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: These comments seem irrelevant here on top of doms. Remove them?

File: src/main/webapp/js/script.js [hunk 45]
```diff
@@ -76,6 +76,128 @@ function showOrHideNotes(version) {
     $('div.setup-panel div a.btn-success').trigger('click');
 });
 
+
+
+
```
Comment (shrutip90) [inline src/main/webapp/js/script.js]: Too many blank lines.

File: src/main/java/firebase/sdk/dashboard/api/PlatformResource.java [hunk 46]
```diff
@@ -0,0 +1,69 @@
...
+   * of the names of all platforms along with the timestamp representing
+   * the closest release deadline.
+   *
+   * @return Response object containing an HTTP status code and a HashMap contaning
+   * the names of all the supported platforms along with a timestamp as its value.
+   */
+  @GET
+  public Response getPlatforms() {
+ 
+      HashMap<String, String> platforms = new HashMap<String, String>();
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/api/PlatformResource.java]: You mentioned that the values in Hashmap are timestamps in the comments. But added just the names. Do you want to update the comments? or add a TODO?

File: src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java [hunk 47]
```diff
@@ -0,0 +1,111 @@
...
+
+    // Delete a release from the Datastore 
+    public void deleteRelease(Release release){
+        Key releaseKey = KeyFactory.createKey("Release", release.platform().getLabel() + "_" + release.releaseName());
+        DATASTORE.delete(releaseKey);
+    }           
+}
+
+
+
```
Comment (shrutip90) [inline src/main/java/firebase/sdk/dashboard/dao/PlatformReleaseDaoDatastore.java]: Too many blank lines.

**Review by shrutip90 — APPROVED** [review]
